cypress/integration/Tools/TestUiViewAndLogs.spec.js

```diff
@@ -0,0 +1,84 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+*/
+
+// -----------------------------------------------------------------------------
+// The Purpose of this Test Spec File is to test the Persistent Logging features
+// and to demonstrate how to skip dependant tests when their predicessor fails.
+// -----------------------------------------------------------------------------
+
+const helpers = require('../../support/Helpers')
+
+// The difference between the 1st and 2nd is where we place the call to SkipRemainingTestsOfSuiteIfFailed()
+describe('1st Test Suite Root', () => {
+  afterEach(helpers.SkipRemainingTestsOfSuiteIfFailed)
+  describe('Level 2-1', () => {
+    it('The FIRST Test', () => {
+      expect(true).to.equal(true)
+    })
+
+    it('The SECOND Test', () => {
+      expect(false).to.equal(true) // Intentionally fail to verify that the next tests will be skipped.
+    })
+
+    it('The THIRD Test', () => {
```

Comment:
Related to our discussion about declaring dependencies.  How would a developer declare this test ('The THIRD Test') to not have dependencies on "The FIRST Test" or "The SECOND Test"?

Say the `before` / `beforeEach` are successful and the test is setup correct. Test 2 depends on Test 1. Test 1 fails so Test 2 does not run, but Test 3 was NOT dependent so Test 3 still runs.

As it looks now it would skip all tests after the failed test.

Reply:
To do that the 3rd test would have to be in a `describe` scope that the 1st and 2nd test is described in and the `SkipRemainingTestsOfSuiteIfFailed` is defined within. So yes this is easily doable. An example of this is in the `2nd Test Suite Root` where the 4th and 5th tests do continue to run even if there is a failure in the 1st, 2nd or 3rd test.

However, in all of the tests I have written so far, all of the subsequent test fragments do depend on the prior tests passing in order to carry on. I doubt there will be many cases that we can come up with where this dependency does not exist.